### PR TITLE
Add PSR7 http-server-middleware

### PR DESCRIPTION
**Pull request by jerv13**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,8 +22,8 @@
   "require": {
     "php": ">=7.0",
     "psr/container": "^1.0",
-    "psr/http-message": "^1.0.1",
-    "reliv/zf-config-factories": "^4",
+    "psr/http-message": "^1.0",
+    "psr/http-server-middleware": "^1.0",
```
Comment (bradharms) [inline composer.json]: I'm guessing this changed from non-semver to semver?

**Comment by rodmcnew** [comment]
Do we really need all these flags just to turn pretty print on and off? Why not just always either have it on or off?

**Review by bradharms — APPROVED** [review]

**Comment by rodmcnew** [comment]
My main concern is just the complexity of having a debug flag passed everywhere. It took me a while to dig up what it did and to ensure it doesn't open some security hole. I'd rather not have it at all. Brad and I chatted some and think we should just never pretty print. Let people use the pretty print feature of their browser or other tools if they need it. Not the server's job.

**Review by bradharms — CHANGES_REQUESTED** [review]
See rod's previous comment.

**Comment by jerv13** [comment]
The 'debug' config is common to expressive and the debug construction arg was already in this code base. I did not add it in this PR.  So is really out of scope for the PR.

https://github.com/zendframework/zend-expressive-skeleton/blob/master/config/autoload/zend-expressive.global.php

**Comment by rodmcnew** [comment]
I'm going to go ahead and merge this regardless of the debug concerns since James explained that code existed previously and was just carried forward to the new expressive version.

**Pull request merged**